### PR TITLE
Feature/cryoclim masks

### PR DESCRIPTION
**Pull request by trygveasp**
Add perm_snow and slope as input to cryoclim as masks

### CODE REVIEW

File: pysurfex/netcdf.py [hunk 1]
```diff
@@ -132,6 +132,7 @@ def nc_slice(
 
         logging.debug("times to read %s", times_to_read)
         levels_to_read = []
+        print(levels)
```
Comment (josteinblyverket) [inline pysurfex/netcdf.py:135]: Remove print?

File: pysurfex/pseudoobs.py [hunk 2]
```diff
@@ -56,7 +66,21 @@ def snow_pseudo_obs_cryoclim(
     iii = 0
     res_lons = []
     res_lats = []
+
+    if neighbourhood:
+        cryo_radius = 2
```
Comment (josteinblyverket) [inline pysurfex/pseudoobs.py:71]: For later some of these settings could enter a config?

File: pysurfex/pseudoobs.py [hunk 3]
```diff
@@ -82,6 +113,40 @@ def snow_pseudo_obs_cryoclim(
     p_fg_elevs = gridpos2points(
         fg_geo.lons, fg_geo.lats, np.asarray(res_lons), np.asarray(res_lats), gelevs_fg
     )
+
+    if neighbourhood:
+        fg_snow_radius = 3
```
Comment (josteinblyverket) [inline pysurfex/pseudoobs.py:118]: Same here

File: pysurfex/pseudoobs.py [hunk 4]
```diff
@@ -119,25 +184,49 @@ def snow_pseudo_obs_cryoclim(
...
+                                if p_snow_fg <= fg_threshold:
+                                    obs_value = p_snow_fg
+                            else:
                                 obs_value = p_snow_fg
                         else:
-                            obs_value = p_snow_fg
-                    else:
-                        obs_value = new_snow_depth
+                            obs_value = new_snow_depth
                 elif snow_class_val == 1:
```
Comment (josteinblyverket) [inline pysurfex/pseudoobs.py:223]: Same here

File: pysurfex/pseudoobs.py [hunk 5]
```diff
@@ -119,25 +184,49 @@ def snow_pseudo_obs_cryoclim(
...
                         laf_threshold,
                     )
+            perm_ok = True
+            if grid_perm_snow is not None:
+                if p_perm_snow[i] > 0.0:
+                    perm_ok = False
+
+            slope_ok = True
+            if grid_slope is not None:
+                if p_slope[i] > 0.5:
```
Comment (josteinblyverket) [inline pysurfex/pseudoobs.py:194]: Later this (0.5) should enter a config?

**Review by josteinblyverket — APPROVED** [review]
Looks OK to me.

File: pysurfex/netcdf.py (hunk 1 above)
Comment (trygveasp) [reply pysurfex/netcdf.py]: Done

File: pysurfex/pseudoobs.py (hunk 2 above)
Comment (trygveasp) [reply pysurfex/pseudoobs.py:71]: Agree

File: pysurfex/pseudoobs.py [hunk 6]
```diff
@@ -119,25 +184,49 @@ def snow_pseudo_obs_cryoclim(
...
+
             # Check if in grid
-            if in_grid[i] and laf_ok:
+            if in_grid[i] and laf_ok and perm_ok and slope_ok:
                 obs_value = np.nan
                 try:
                     snow_class_val = p_snow_class[str(i)]
                 except KeyError:
                     continue
                 if snow_class_val == 2:
```
[the remark this answers is not in the document]
Comment (trygveasp) [reply pysurfex/pseudoobs.py:204]: Not used if not given as input

File: pysurfex/pseudoobs.py (hunk 5 above)
Comment (trygveasp) [reply pysurfex/pseudoobs.py:194]: Agree

**Pull request merged**